tests/test_appliancesmanager.py

```diff
@@ -1,123 +1,148 @@
-    mock_appliancesmanager_get_shared_appliances_get,
-)
+CURR_DIR = Path(__file__).parent
+DATA_DIR = CURR_DIR / "data"
+ACCOUNT_ID = "12345"
 
-BACKEND_SELECTOR_MOCK = BackendSelectorMock()
 
+def get_mock_coro(return_value):
+    async def mock_coro(*args, **kwargs):
+        return return_value
 
-def assert_appliances_manager_call(
-    http_client_mock: AiohttpClientMocker,
-    call_index: int,
-    path: str,
-    required_headers: dict = None,
-    forbidden_headers: list = None,
-):
-    mock_calls = http_client_mock.mock_calls
+    return Mock(wraps=mock_coro)
 
-    call = mock_calls[call_index]
-    assert call[0] == "GET"
-    assert BACKEND_SELECTOR_MOCK.base_url + call[1].path == path
-    # call[2] is body, which will be None
 
-    if required_headers is not None:
-        for k, v in required_headers.items():
-            assert call[3][k] == v
+with open(DATA_DIR / "owned_appliances.json") as f:
+    owned_appliance_data = json.load(f)
 
-    if forbidden_headers is not None:
-        for k in forbidden_headers:
-            assert k not in call[3]
+with open(DATA_DIR / "shared_appliances.json") as f:
+    shared_appliance_data = json.load(f)
 
 
-@pytest.mark.parametrize("account_id", [None, ACCOUNT_ID])
-async def test_fetch_appliances_with_set_account_id(
-    account_id: str, http_client_mock: AiohttpClientMocker
-):
-    get_appliances_idx = 0 if account_id is not None else 1
+@pytest.fixture(autouse=True)
+def account_id_calls_fixture(aioresponses_mock, backend_selector_mock):
+    """Mock the calls to get the user details and owned appliances.
 
-    mock_appliancesmanager_get_account_id_get(http_client_mock, BACKEND_SELECTOR_MOCK)
-    mock_appliancesmanager_get_owned_appliances_get(
-        http_client_mock, BACKEND_SELECTOR_MOCK, ACCOUNT_ID
+    This is required by all of the tests in this file, so we use the autouse=True
+    parameter to ensure that this fixture is always run.
+    """
+    aioresponses_mock.get(
+        backend_selector_mock.user_details_url, payload={"accountId": ACCOUNT_ID}
+    )
+    aioresponses_mock.get(
+        backend_selector_mock.get_owned_appliances_url(ACCOUNT_ID),
+        payload={ACCOUNT_ID: owned_appliance_data},
     )
-    mock_appliancesmanager_get_shared_appliances_get(
-        http_client_mock, BACKEND_SELECTOR_MOCK
+    aioresponses_mock.get(
+        backend_selector_mock.shared_appliances_url, payload=shared_appliance_data
     )
+    yield
 
-    http_client_mock.create_session(asyncio.get_event_loop())
-    auth = Auth(BACKEND_SELECTOR_MOCK, "email", "secretpass", http_client_mock.session)
 
-    if account_id is not None:
-        auth._auth_dict["accountId"] = account_id
+async def test_uer_details_not_called_if_account_id_is_known(
+    auth_mock: Auth,
+    backend_selector_mock: BackendSelector,
+    appliances_manager: AppliancesManager,
+    aioresponses_mock,
+):
+    auth_mock._auth_dict["accountId"] = ACCOUNT_ID
```

Comment:
can we have this test without changing internal stuff of the class?
Maybe by:
- calling `fetch_appliances()` once
- checking the request was done
- resetting `aioresponses_mock.requests`
- calling `fetch_appliances()` again
- checking that the request was not done
 
If that works, we should probably rename the test to `test_user_details_requested_only_once` or something, and can remove the `test_user_details_called_if_account_id_not_known` test.

Reply:
Good idea, I've implemented this test and removed the other one